packages/runtime-core/src/componentSlots.ts

```diff
@@ -22,16 +22,37 @@ import { isHmrUpdating } from './hmr'
 import { DeprecationTypes, isCompatEnabled } from './compat/compatConfig'
 import { toRaw } from '@vue/reactivity'
 
-export type Slot = (...args: any[]) => VNode[]
+export type Slot = (...args: any[]) => VNode[] | VNode
```

Comment:
`h(Foo, () => {...}` can return a single VNode but `slots.default()` is normalised so should always be `VNode[]`